web/src/containers/activity/CostAllocate.test.js

```diff
@@ -33,48 +33,44 @@ describe('the CostAllocate component', () => {
 
   test('updates activity when text is changed', () => {
     const component = shallow(<CostAllocate {...props} />);
+    const mockedEvent = 
 
     component
       .find('RichText')
       .filterWhere(n => n.props().content === 'cost allocation')
-      .simulate('sync', 'bloop');
-    expect(
-      props.updateActivity.calledWith('activity id', {
-        costAllocationDesc: 'bloop'
-      })
-    );
+      .prop('onSync')('bloop');
+    expect(props.setMethodology).toHaveBeenCalledWith(1, 'bloop');
 
     component
       .find('RichText')
       .filterWhere(n => n.props().content === 'other funding')
-      .simulate('sync', 'florp');
-    expect(
-      props.updateActivity.calledWith('activity id', {
-        otherFundingDesc: 'florp'
-      })
-    );
+      .prop('onSync')('florp');
+    expect(props.setOtherFunding).toHaveBeenCalledWith(1, 'florp');
   });
 
   test('maps redux state to component props', () => {
-    expect(
-      mapStateToProps(
-        {
-          activities: {
-            byKey: {
-              key: 'this is the activity'
+    const state = {
+      apd: {
+        data: {
+          activities: [
+            {
+              key: 'activity key'
             }
-          }
-        },
-        { aKey: 'key' }
-      )
-    ).toEqual({
-      activity: 'this is the activity'
+          ]
+        }
+      }
+    };
+    expect(mapStateToProps(state, {activityIndex: 1})).toEqual({
+        activity: {
+          key: 'activity key'
```

Comment:
@mgwalker I'm not sure why this isn't passing; I think I've got the mock props formatted wrongly? I'm getting this error:
```
 expect(received).toEqual(expected) // deep equality

    - Expected
    + Received

      Object {
    -   "activity": Object {
    -     "key": "activity key",
    -   },
    +   "activity": null,
      }
```

Reply:
In `state` starting on line 52, there's only one activity, so when you ask for activity index 1, there's nothing there.

Reply:
😂 😭 

I swear I've worked with computers before.